### PR TITLE
Added release notes topic.

### PR DESCRIPTION
**Pull request by debadair**
Created a Release Notes topic in the Elasticsearch reference. Putting the release notes in the docs will make it easier for users to find them and make them easier to maintain.

### CODE REVIEW

**Comment by debadair** [comment]
Hold off reviewing/commenting--adding the release notes for the pre-GA versions.

File: docs/reference/release-notes/release-notes-2_0_0.asciidoc [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+=== Bug Fixes
+
+CRUD:
+
+* Send response for update request when it timed out {pr}/13101[#13101]
+
+Engine:
+
+* Never wrap searcher for internal engine operations {pr}/14071[#14071]
+Fielddata:
```
Comment (clintongormley) [inline docs/reference/release-notes/release-notes-2_0_0.asciidoc]: Need  a blank line before and after this line

File: docs/reference/release-notes/release-notes-2_0_0.asciidoc [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+[float]
+=== Upgrades
+
+Network:
+
+* Upgrade Netty to 3.10.5.final {pr}/14105[#14105]
+
+[float]
+=== Build
```
Comment (clintongormley) [inline docs/reference/release-notes/release-notes-2_0_0.asciidoc]: For the release notes, I exclude everything from Build onwards (ie Build, Test, Docs, Not classified etc)

File: docs/reference/release-notes/release-notes-2_0_0.asciidoc [hunk 3]
```diff
@@ -0,0 +1,143 @@
+[[release-notes-2_0_0]]
+== 2.0.0 Release Notes
+
+[float]
```
Comment (clintongormley) [inline docs/reference/release-notes/release-notes-2_0_0.asciidoc]: I'd include a link to the github issues search here

**Comment by clintongormley** [comment]
Release notes added to 2.* docs

**Pull request closed without merging**